nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Updater/PackageReferenceUpdater.cs

```diff
@@ -60,45 +63,67 @@ public static async Task UpdateDependencyAsync(
         {
-            return;
+            return [];
         }
 
+        var updateOperations = new List<UpdateOperationBase>();
         var peerDependencies = await GetUpdatedPeerDependenciesAsync(repoRootPath, projectPath, tfms, dependencyName, newDependencyVersion, experimentsManager, logger);
         if (experimentsManager.UseLegacyDependencySolver)
         {
             if (isTransitive)
             {
-                await UpdateTransitiveDependencyAsync(repoRootPath, projectPath, dependencyName, newDependencyVersion, buildFiles, experimentsManager, logger);
+                var updatedFiles = await UpdateTransitiveDependencyAsync(repoRootPath, projectPath, dependencyName, newDependencyVersion, buildFiles, experimentsManager, logger);
+                updateOperations.Add(new PinnedUpdate()
+                {
+                    DependencyName = dependencyName,
+                    NewVersion = NuGetVersion.Parse(newDependencyVersion),
+                    UpdatedFiles = [.. updatedFiles],
+                });
             }
             else
             {
                 if (peerDependencies is null)
                 {
-                    return;
+                    return updateOperations;
                 }
 
-                await UpdateTopLevelDepdendency(repoRootPath, buildFiles, tfms, dependencyName, previousDependencyVersion, newDependencyVersion, peerDependencies, experimentsManager, logger);
+                var topLevelUpdateOperations = await UpdateTopLevelDepdendency(repoRootPath, buildFiles, tfms, dependencyName, previousDependencyVersion, newDependencyVersion, peerDependencies, experimentsManager, logger);
+                updateOperations.AddRange(topLevelUpdateOperations);
             }
         }
         else
         {
             if (peerDependencies is null)
             {
-                return;
+                return updateOperations;
             }
 
-            await UpdateDependencyWithConflictResolution(repoRootPath, buildFiles, tfms, projectPath, dependencyName, previousDependencyVersion, newDependencyVersion, isTransitive, peerDependencies, experimentsManager, logger);
+            var conflictResolutionUpdateOperations = await UpdateDependencyWithConflictResolution(
+                repoRootPath,
+                buildFiles,
+                tfms,
+                projectPath,
+                dependencyName,
+                previousDependencyVersion,
+                newDependencyVersion,
+                isTransitive,
+                peerDependencies,
+                experimentsManager,
+                logger);
+            updateOperations.AddRange(conflictResolutionUpdateOperations);
         }
 
         if (!await AreDependenciesCoherentAsync(repoRootPath, projectPath, dependencyName, buildFiles, tfms, experimentsManager, logger))
         {
-            return;
+            // should we return an empty set because we failed?
+            return updateOperations;
```

Comment:
This will at least tell us what type of operation we failed on right?

Reply:
No added information to this part yet, it's just an early exit because the attempted update resulted in a package downgrade warning.  This will likely be future work to better report why something failed.  Currently I'm just trying to report on why something succeeded and what it did.